docs/reference/verify-assets.mdx

```diff
@@ -106,11 +109,68 @@ As the SBOM was signed with our private key, you can now inspect it to track dow
 
 ## Verify image-based gadgets
 
-### With `ig`
-
 Like our container image, we sign all our image-based gadgets.
-The signature are verified by default using Inspektor Gadget public key:
+The signature are verified by default using Inspektor Gadget public key.
+
+<Tabs groupId="env">
+<TabItem value="kubectl-gadget" label="kubectl gadget">
+Compared to `ig`, you cannot specify information with regard to verification when calling `kubectl gadget run`:
+
+```bash
+$ kubectl gadget run --public-keys="$(cat your-key.pub)" trace_exec
+Error: unknown flag: --public-keys
+$ kubectl gadget run --verify-image=false trace_exec
+Error: unknown flag: --verify-image
+```
+
+Instead, all these information are set once at deploy time.
+This avoid them being tampered by any users.
+By default all image-based gadgets are verified using Inspektor Gadget public key:
+
+```bash
+$ kubectl gadget deploy
+...
+Inspektor Gadget successfully deployed
+$ kubectl gadget run ghcr.io/your_repo/trace_exec -A
+Error: fetching gadget information: getting gadget info: rpc error: code = Unknown desc = getting gadget info: initializing and preparing operators: instantiating operator "oci": ensuring image: verifying image "ghcr.io/your_repo/trace_exec": verifying signature: invalid signature when validating ASN.1 encoded signature
+$ kubectl gadget run trace_exec -A
+K8S.NAMESP… K8S.PODNAME K8S.CONTAI… MNTNS… TIMES… PID    PPID   UID    GID    LOGIN… SESSI… RETVAL ARGS_… UPPER… ARGS_… COMM  ARGS  K8S.…
+gadget      gadge…hbh8n gadget      40265… 22867… 53386  53368  0      0      42949… 42949… 0      2      false  35     gadg… /bin… mini…
+gadget      gadge…hbh8n gadget      40265… 22867… 53387  53369  0      0      42949… 42949… 0      2      false  35     gadg… /bin… mini…
+...
+```
+
+You can specify a custom public key at deploy time using `--gadgets-public-keys`:
+
+```bash
+$ kubectl gadget deploy --gadgets-public-keys="$(cat your-key.pub)"
+...
+Inspektor Gadget successfully deployed
+$ kubectl gadget run trace_exec -A
```

Comment:
Maybe highlight that this _should_ fail, otherwise it might confuse. Maybe also first do the positive test (running _their_ signed gadget) and then the negative one.

```suggestion
# try to run an official gadget (notice that we didn't include the official Inspektor Gadget public key anymore)
$ kubectl gadget run trace_exec -A
```